scala_proto/scala_proto.bzl

```diff
@@ -406,14 +406,19 @@ def _gen_proto_srcjar_impl(ctx):
 
     deps_jars = collect_jars(jvm_deps)
 
-    worker_content = "{output}\n{paths}\n{flags_arg}\n{packages}".format(
+    blacklisted_protos = [dep for target in ctx.attr.blacklisted_protos for dep in target.proto.transitive_sources]
+    blacklisted_protos_dict = dict(zip(blacklisted_protos, blacklisted_protos))
+    inputs = [f.path for f in sorted(acc_imports) if blacklisted_protos_dict.get(f, None) == None]
```

Comment:
I think this `sorted` can be expensive. WDYT about only doing this whole line if `blacklisted_protos` is non empty?
also wdyt about using `if blacklisted_protos_dict.has_key(f)`?
lastly maybe it's best to just do:
` inputs = [f.path for f in retained_protos(acc_imports, blacklisted_protos_dict)`?
then the if and the details can be kept out of this context?

Reply:
type 'dict' has no method has_key(File).
I'll extract to separate function. 
Not sure I understood your comment about sorting. If blacklisted_protos would be empty I would still need to sort that list. I don't think sorting is needed at all. I just followed pattern from `_colon_paths`